java-filter-chain/src/main/java/org/wso2/mgw/filterchain/JWTValidator/JWTValidator.java

```diff
@@ -81,11 +110,38 @@ public static boolean validateSignature(String jwtToken, String signature){
         JWTClaimsSet payload = null;
         SignedJWT parsedJWTToken;
         boolean isVerified = false;
-        try{
-            parsedJWTToken = (SignedJWT) JWTParser.parse(jwtToken);
-            isVerified = verifyTokenSignature(parsedJWTToken);
-        }catch (ParseException e) {
-            System.out.println("Invalid JWT token. Failed to decode the token.");
+        try {
+            if (enableCache) {
+                if(GatewayApiKeyCache.get(signature) != JWTConstants.UNAVAILABLE){
+                    System.out.println("Api Key retrieved from the Api Key cache.");
```

Comment:
Lets remove the logs(sout), in order to run the perf tests. Check other places as well